server/s3_get_bucket_replication_action.cc

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#include "s3_get_bucket_replication_action.h"
+#include "s3_error_codes.h"
+#include "s3_log.h"
+
+S3GetBucketReplicationAction::S3GetBucketReplicationAction(
+    std::shared_ptr<S3RequestObject> req,
+    std::shared_ptr<S3BucketMetadataFactory> bucket_meta_factory)
+    : S3BucketAction(std::move(req), std::move(bucket_meta_factory), false) {
+  s3_log(S3_LOG_DEBUG, request_id, "%s Ctor\n", __func__);
+
+  s3_log(S3_LOG_INFO, stripped_request_id,
+         "S3 API: Get Bucket Replication. Bucket[%s]\n",
+         request->get_bucket_name().c_str());
+
+  setup_steps();
+}
+
+void S3GetBucketReplicationAction::setup_steps() {
```

Comment:
I think your code would generally be improved with a lot more comments.

Reply:
All the action files has almost common format. That's why there are not much comments in action file.
I tried to follow other API's action file.
Code for each API is different in ..body.cc file. 
I have added appropriate comments there. 
If necessary I can add comments here too.